quil-py/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+use pyo3::prelude::*;
+use rigetti_pyo3::create_init_submodule;
+
+pub mod instruction;
+pub mod program;
+
+create_init_submodule! {
+    submodules: [ "instructions": instruction::init_submodule, "program": program::init_submodule ],
+}
+
+#[pymodule]
+fn quil(py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    init_submodule("quil", py, m)?;
+    Ok(())
+}
+
+pub fn init_quil_submodule(name: &str, py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    init_submodule(name, py, m)?;
+    Ok(())
+}
```

Comment:
This is what allows us to re-export these binding through `qcs-sdk-python`. One unanswered question, how will type hints work with this approach?

Reply:
Yeah I think a combination of workspace dependencies and specifying a specific version of `quil` as a dependency of the `qcs-sdk-python` package will allow us to enforce consistent `quil` versions.